v-next/core/test/plugins/plugin-validation.ts

```diff
@@ -0,0 +1,115 @@
+import assert from "node:assert/strict";
+import { describe, it } from "node:test";
+
+import { HardhatError } from "@nomicfoundation/hardhat-errors";
+
+import { validatePluginNpmDependencies } from "../../src/internal/plugins/plugin-validation.js";
+import { HardhatPlugin } from "../../src/types/plugins.js";
+
+describe("Plugins - plugin validation", () => {
+  const plugin: HardhatPlugin = {
+    id: "example-plugin",
+    npmPackage: "example",
+  };
+
+  it("should skip validation if the plugin is not from an npm package", async () => {
+    await assert.doesNotReject(async () =>
+      validatePluginNpmDependencies({
+        ...plugin,
+        npmPackage: undefined,
+      }),
+    );
+  });
+
+  describe("when the plugin has no peer deps", () => {
+    it("should pass validation if the npm package has been installed", async () => {
+      const installedPackageProjectFixture = import.meta.resolve(
+        "./fixture-projects/installed-package",
+      );
+
+      await assert.doesNotReject(async () =>
+        validatePluginNpmDependencies(plugin, installedPackageProjectFixture),
+      );
+    });
+
+    it("should fail validation if the npm package has not been installed", async () => {
+      const noninstalledPackageProjectFixture = import.meta.resolve(
+        "./fixture-projects/not-installed-package",
+      );
+
+      await assert.rejects(
+        async () =>
+          validatePluginNpmDependencies(
+            plugin,
+            noninstalledPackageProjectFixture,
+          ),
+        (err) => {
+          assert(HardhatError.isHardhatError(err), "Expected a HardhatError");
+          assert.equal(
+            err.message,
+            'HHE1200: Plugin "example-plugin" is not installed.',
+          );
+
+          return true;
+        },
+        "Expected a plugin not installed error",
+      );
+    });
+  });
+
+  describe("when the plugin has peer deps", () => {
+    it("should pass validation if the peer deps have been installed", async () => {
+      const installedPeerDepsFixture = import.meta.resolve(
+        "./fixture-projects/installed-peer-deps",
+      );
+
+      await assert.doesNotReject(async () =>
+        validatePluginNpmDependencies(plugin, installedPeerDepsFixture),
+      );
+    });
+
+    it("should fail validation if a peer dependency is not installed", async () => {
+      const notInstalledPeerDepFixture = import.meta.resolve(
+        "./fixture-projects/not-installed-peer-dep",
+      );
+
+      await assert.rejects(
+        async () =>
+          validatePluginNpmDependencies(plugin, notInstalledPeerDepFixture),
+        (err) => {
+          assert(HardhatError.isHardhatError(err), "Expected a HardhatError");
+          assert.equal(
+            err.message,
+            'HHE1201: Plugin "example-plugin" is missing a peer dependency "peer2".',
+          );
+
+          return true;
+        },
+        "Expected a missing peer dependency error",
+      );
```

Comment:
You can simplify this:
```suggestion
      await assert.rejects(
        validatePluginNpmDependencies(plugin, notInstalledPeerDepFixture),
        {
          name: "HardhatError"
          message: 'HHE1201: Plugin "example-plugin" is missing a peer dependency "peer2".',
        }
      );
```

Reply:
Oh that is nice.